### PR TITLE
Use configuration.getPassword() if available to eliminate password from druid.xml

### PR DESCRIPTION
**Pull request by hddong**
It is not safe that put a plaintext password in druid.xml, use hadoop Configuration.getPassword instead.

How to use:
1. use `hadoop credential create smart.druid.password -provider jceks://file/${some-path}/smart-site.jceks` to create a jecks file.
2. delete password form druid.xml
3. add 
```
<property>
        <name>hadoop.security.credential.provider.path</name>
        <value>jceks://file/${path of jceks}/smart-site.jceks</value>
    </property>
```
in conf/smart-default.xml
4. restart all

### CODE REVIEW

**Comment by hddong** [comment]
reset this pls

File: smart-metastore/src/main/java/org/smartdata/metastore/utils/MetaStoreUtils.java [hunk 1]
```diff
@@ -510,6 +512,16 @@ public static MetaStore getDBAdapter(
...
         }
 
+        try {
+          String pw = HadoopUtils
+            .getPasswordFromHadoop(SmartConfKeys.SMART_DRUID_PASSWORD, conf);
+          if (pw != null && pw != "") {
+            p.setProperty("password", pw);
+          }
+        } catch (IOException e) {
+          LOG.info("Can not get password from hadoop, use default set.");
```
Comment (philo-he) [inline smart-metastore/src/main/java/org/smartdata/metastore/utils/MetaStoreUtils.java]: Suggestion: LOG.warn("Can not get metastore password from hadoop provision credentials, use the one configured in druid.xml");

File: smart-common/src/main/java/org/smartdata/conf/SmartConfKeys.java [hunk 2]
```diff
@@ -48,6 +48,7 @@
   public static final String SMART_NAMESPACE_FETCHER_CONSUMERS_NUM_KEY =
       "smart.namespace.fetcher.consumers.num";
   public static final int SMART_NAMESPACE_FETCHER_CONSUMERS_NUM_DEFAULT = 3;
+  public static final String SMART_DRUID_PASSWORD = "smart.druid.password";
```
Comment (philo-he) [inline smart-common/src/main/java/org/smartdata/conf/SmartConfKeys.java]: Suggestion: smart.metastore.password

Comment (philo-he) [reply smart-common/src/main/java/org/smartdata/conf/SmartConfKeys.java]: It would be better if a comment is put above.

**Comment by philo-he** [comment]
Thanks @hddong for your patch. This is an awesome thought. After a quick review, I posted some comments. You can put that property in smart-default.xml and give a description. Hope I am not missing something. Thanks again!

**Comment by hddong** [comment]
@PHILO-HE thanks for your suggestions, all of them done.

**Comment by philo-he** [comment]
@hddong, thanks for your contribution. This patch basically looks good on hadoop-3.1.0. But I need to verify it further on cdh to see whether there is compatibility issue.

**Comment by hddong** [comment]
@PHILO-HE I'm glad to hear it worked correct on hadoop-3.1.0. Actully, I have tested only on hadoop-3.1.0.
By the way,  why test always throw `java.lang.AssertionError: expected:<COMPRESSION> but was:<NORMAL>` , Is there any way to solve it？

**Comment by philo-he** [comment]
@hddong, please ignore that AssertionError. It is related to state inconsistency for compression action. I will fix it. Also, I am doing some tests on hadoop-2.7.3 and CDH hadoop-2.6.0. 
Another suggestion: it looks better if getPasswordFromHadoop is moving to SmartConf.java, thus, other classes can directly request password by SmartConf, which makes more sense. Thanks!

**Comment by hddong** [comment]
> @hddong, please ignore that AssertionError. It is related to state inconsistency for compression action. I will fix it. Also, I am doing some tests on hadoop-2.7.3 and CDH hadoop-2.6.0.
> Another suggestion: it looks better if getPasswordFromHadoop is moving to SmartConf.java, thus, other classes can directly request password by SmartConf, which makes more sense. Thanks!

Thanks for your check, had move it to SmartConf.java.

**Comment by philo-he** [comment]
The patch has been merged to trunk. Thanks @hddong for your contribution! I will update the relevant doc in some other PR.

**Comment by philo-he** [comment]
@hddong, any other patch is welcome! And if you need, please feel free to discuss this project with me.

**Comment by hddong** [comment]
@PHILO-HE thanks for your review and suggestions.

**Pull request merged**